OpenRA.Mods.Common/Scripting/Global/ActorGlobal.cs

```diff
@@ -71,7 +71,20 @@ public int BuildTime(string type)
 			if (!Context.World.Map.Rules.Actors.TryGetValue(type, out ai))
 				throw new LuaException("Unknown actor type '{0}'".F(type));
 
-			return ai.GetBuildTime();
+			var bi = ai.TraitInfoOrDefault<BuildableInfo>();
+
+			if (bi == null)
+				return 0;
+
+			var time = bi.BuildDuration;
+			if (time == -1)
+			{
+				var valued = ai.TraitInfoOrDefault<ValuedInfo>();
+				time = valued != null ? valued.Cost : 0;
+			}
+
+			time = time * bi.BuildDurationModifier / 100;
```

Comment:
It's not really acceptable to have a Lua API that returns demonstrably wrong values when using traits in a supported way, so this will beed to be resolved in *some* way.  You could look up the default queue type and include an optional param to override if there are multiple.